implementation/contracts/deposit/DepositFunding.sol

```diff
@@ -103,6 +104,14 @@ library DepositFunding {
             block.timestamp > _d.signingGroupRequestedAt.add(TBTCConstants.getSigningGroupFormationTimeout()),
             "Signing group formation timeout not yet elapsed"
         );
+
+        // refund the deposit owner the cost to create a new keep at current price levels.
```

Comment:
I believe the comment from this function docs `/// @dev We rely on the keep system to punish the signers in this case.` is no longer valid.

Reply:
Do keeps do no punishment on DKG failure? Guessing not because it's not attributable.

Reply:
Currently not, signers are punished on proved signature fraud.

In the case of DKG failure signers' bonds can be seized by the keep owner. Punishing the clients and exposing dedicated function for fee refund will be considered in https://github.com/keep-network/keep-ecdsa/issues/324.